Gems/AzQtComponentsForPython/CMakeLists.txt

```diff
@@ -0,0 +1,224 @@
+                    COMMENT "Running generator for ${typesystem_file}.")
+    
+# Create dummy wrapper files so the second project has something to work with while being created.
+include(wrapped_files.cmake)
+foreach(WRAPPER_FILE ${WRAPPED_FILES})
+    if(NOT EXISTS "${WRAPPER_FILE}")
+        file(WRITE ${WRAPPER_FILE} "")
+    endif()
+endforeach()
+    
+# Set up the project for building the code generated by shiboken.
+ly_add_target(
+    NAME ${BINDINGS_LIBRARY} SHARED
+    NAMESPACE Gem
+    FILES_CMAKE
+        wrapped_files.cmake
+    INCLUDE_DIRECTORIES
+        PRIVATE
+            ${PYSIDE2_INCLUDE_DIR}
+            ${PYTHON_INCLUDE_DIR}
+            ${SHIBOKEN_INCLUDE_DIR}
+            "${PYSIDE2_INCLUDE_DIR}\\QtCore"
+            "${PYSIDE2_INCLUDE_DIR}\\QtGui"
+            "${PYSIDE2_INCLUDE_DIR}\\QtWidgets"
+            "${PYSIDE2_INCLUDE_DIR}\\QtNetwork"
+    BUILD_DEPENDENCIES
+        PUBLIC
+            Qt5::Widgets
+            Qt5::Core
+            Qt5::Widgets
+            Qt5::Gui
+            ${AZQTCOMPONENTS_LIBRARY}
+            ${BINDINGS_LIBRARY}.bindings
+)
+
+target_link_libraries( ${BINDINGS_LIBRARY} PRIVATE 
+    debug ${PYSIDE2_SHARED_LIBRARIES_D}
+    optimized ${PYSIDE2_SHARED_LIBRARIES}
+)
+
+target_link_libraries( ${BINDINGS_LIBRARY} PRIVATE 
+    debug ${SHIBOKEN_SHARED_LIBRARIES_D}
+    optimized ${SHIBOKEN_SHARED_LIBRARIES}
+)
+
+target_sources(${BINDINGS_LIBRARY} PRIVATE ${WRAPPED_FILES})
+
+# Building the wrapper files will not work with a unity build.
+set_target_properties(${BINDINGS_LIBRARY} PROPERTIES UNITY_BUILD OFF)
+
+# Append _d to the module name in a debug build.
+set_target_properties(${BINDINGS_LIBRARY} PROPERTIES DEBUG_POSTFIX "_d")
+
+# Disable various warnings in shiboken generated wrapper code.
+#"conditional expression is constant"
+#"unreferenced formal parameter"
+#"declaration of 'x' hides previous local declaration."
+#"declaration of 'x' hides class member."
+target_compile_options(${BINDINGS_LIBRARY} PRIVATE /wd4127 /wd4100 /wd4456 /wd4458)
+
+# Fix the name of the module.
+set_property(TARGET ${BINDINGS_LIBRARY} PROPERTY PREFIX "")
+set_property(TARGET ${BINDINGS_LIBRARY} PROPERTY OUTPUT_NAME
+             "${BINDINGS_LIBRARY}${PYTHON_EXTENSION_SUFFIX}")
+set_property(TARGET ${BINDINGS_LIBRARY} PROPERTY SUFFIX ".pyd")
+
+# Ensure linking doesn't happen against the wrong libraries during #pragma comment(lib)
+if (CMAKE_BUILD_TYPE STREQUAL "debug")
+    set(python_additional_link_flags "/NODEFAULTLIB:\"python37_d.lib\"")
+    set(python_additional_link_flags ${python_additional_link_flags} "/NODEFAULTLIB:\"python37_d.lib\"")
+else()
+    set(python_additional_link_flags "/NODEFAULTLIB:\"python37.lib\"")
+    set(python_additional_link_flags ${python_additional_link_flags} "/NODEFAULTLIB:\"python37.lib\"")
+endif()
+
+set_target_properties(${BINDINGS_LIBRARY} PROPERTIES LINK_FLAGS "${python_additional_link_flags}")
+
+
+        
```

Comment:
nitpick: There is trailing whitespace on the this line

Reply:
Fixed